File: examples/projects/benchmarks/region_alloc/region_alloc.cpp
// 基准: region_alloc —— 100 万次小对象分配 + 释放（new/delete）
// 与 region_alloc.zeta 逻辑严格一致。输出 = 499500000
// 注: Zeta 侧用 region 批量分配（区域退出一次释放），本侧为每次 new+delete。
#include <cstdio>
#include <cstdint>

struct Big {
    int64_t a, b, c, d;
};

int main() {
    int64_t sum = 0;
    for (int i = 0; i < 1000000; i++) {
        Big *o = new Big{i % 1000, i, i * 2, i * 3};
        sum += o->a;
        delete o;
    }
    printf("%lld\n", sum);
    return 0;
}

File: examples/projects/benchmarks/region_alloc/region_alloc.swift
// 基准: region_alloc —— 100 万次小对象分配（class 堆分配 + ARC 释放）
// 与 region_alloc.rl 逻辑严格一致。输出 = 499500000
// 注: Rlyeh 侧用 region 批量分配（区域退出一次释放），本侧为 class 实例化 + ARC。
final class Big {
    let a: Int64, b: Int64, c: Int64, d: Int64
    init(_ a: Int64, _ b: Int64, _ c: Int64, _ d: Int64) {
        self.a = a; self.b = b; self.c = c; self.d = d
    }
}

var sum: Int64 = 0
for i in 0..<1_000_000 {
    let o = Big(Int64(i % 1000), Int64(i), Int64(i * 2), Int64(i * 3))
    sum += o.a
}
print(sum)
